src/Broadway/CommandHandling/ClosureCommandHandler.php

```diff
@@ -31,13 +32,26 @@ public function add(\Closure $handler): void
         $reflection = new \ReflectionFunction($handler);
         $reflectionParams = $reflection->getParameters();
 
-        if (!isset($reflectionParams[0]) || !$reflectionParams[0]->getClass()) {
+        if (0 === $reflection->getNumberOfParameters()) {
             throw new ClosureParameterNotAnObjectException();
         }
 
-        $index = $reflectionParams[0]->getClass()->getName();
+        $reflectionType = $reflectionParams[0]->getType();
 
-        $this->handlers[$index] = $handler;
+        $name = null;
+        if ($reflectionType instanceof \ReflectionNamedType) {
+            /** @var \ReflectionNamedType $reflectionNamed */
+            $reflectionNamed = $reflectionParams[0]->getType();
```

Comment:
This is duplicate code and still not very readable. Can you simplify it based on my earlier suggestion?

Reply:
Ok, sure!